src/chrome/content/scaffold/scaffold.xul

```diff
@@ -156,10 +157,12 @@
 								</menupopup>
 							</toolbarbutton>
 						</hbox>
-						<iframe src="data:text/html,%3C!DOCTYPE%20html%20PUBLIC%20%22-//W3C//DTD%20XHTML%201.1//EN%22%20%22http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd%22%3E%3Chtml%20xmlns=%22http://www.w3.org/1999/xhtml%22%3E%3Chead%3E%3Cscript%20type=%22text/javascript%22%20src=%22chrome://scaffold/content/ace/ace.js%22%3E%3C/script%3E%3Cscript%20type=%22text/javascript%22%20src=%22chrome://scaffold/content/ace/mode-javascript.js%22%3E%3C/script%3E%3Cscript%20type=%22text/javascript%22%20src=%22chrome://scaffold/content/aceWrapper.js%22%3E%3C/script%3E%3C/head%3E%3Cbody%20id=%22body%22%3E%3C/body%3E%3C/html%3E" id="editor-code" flex="1"/>
+						<iframe src="data:text/html,%3C!DOCTYPE%20html%20PUBLIC%20%22-//W3C//DTD%20XHTML%201.1//EN%22%20%22http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd%22%3E%3Chtml%20xmlns=%22http://www.w3.org/1999/xhtml%22%3E%3Chead%3E%3Cscript%20type=%22text/javascript%22%20src=%22chrome://scaffold/content/ace/ace.js%22%3E%3C/script%3E%3Cscript%20type=%22text/javascript%22%20src=%22chrome://scaffold/content/ace/mode-javascript.js%22%3E%3C/script%3E%3Cscript%20src=%22chrome://scaffold/content/ace/ext-language_tools.js%22%3E%3C/script%3E%3Cscript%20type=%22text/javascript%22%20src=%22chrome://scaffold/content/aceWrapper.js%22%3E%3C/script%3E%3C/head%3E%3Cbody%20id=%22body%22%3E%3C/body%3E%3C/html%3E" id="editor-code" flex="1"/>
 						<hbox id="editor-external-box" align="center">
 							<checkbox id="checkbox-editor-external"/>
 							<label class="label-metadata" value="&scaffold.editor.external.label;" control="checkbox-editor-external"/>
+							<checkbox id="checkbox-live-autocompletion"/>
+							<label class="label-metadata" value="&scaffold.editor.liveAutocompletion.label;"/>
```

Comment:
I think this can just be on rather than configurable. If someone doesn't want it, they're probably not editing in ACE to begin with.

Reply:
There is also the basic autocompletion which is invoked with ctrl+space (at least in windows) which is always activated.

The live autocompletion is IMO sometimes a little disturbing, e.g. I type `a` in the editor and get suggestions for `ZU.cleanAuthor`, `Z.monitorDOMChanges`, `ZU.xpath` etc. Try it out yourself at http://plnkr.co/edit/Ep3bVPjjkm1laImq443t?p=preview .

Moreover, any feedback from @adam3smith would be appreciated about this.

Reply:
Yes, I've tried it. If we're going to have it at all, I think it's more awkward to have to turn it on and off.

Reply:
But also that just sounds like bad autocomplete. It shouldn't trigger `ZU.cleanAuthor` when you type `a`. Can we make it left-bound?

Reply:
Make it left-bound or wait for at least 2-3 characters could work better, but I haven't found anything like this (but it is very hard to find information at all).

The live autocompletion is also disabled in the official demo https://ace.c9.io/build/kitchen-sink.html

Reply:
Minimum length is a good idea. Assuming we can get to them from the outside, I suspect we can monkey patch `getCompletionPrefix` or `retrievePrecedingIdentifier` to do that.

Reply:
They may even take such a patch into their official repo, which we could then simply reuse the official way. However, that is not something I can do currently.

Reply:
That's fine — I can look into this. But I think we should assume we'll have less annoying autocomplete and not bother with a checkbox for it.